$DragonFly: src/gnu/usr.bin/groff/src/libs/libgroff/patches/cset.cpp.patch,v 1.1 2005/09/01 20:47:13 asmodai Exp $

--- cset.cpp	2005-08-31 14:28:07.000000000 +0200
+++ cset.cpp	2005-08-31 15:53:50.000000000 +0200
@@ -19,6 +19,9 @@
 Foundation, 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. */
 
 #include <ctype.h>
+#if defined(__DragonFly__) || defined(__FreeBSD__)
+#include <locale.h>
+#endif
 #include "cset.h"
 
 cset csalpha(CSET_BUILTIN);
@@ -33,7 +36,7 @@
 cset csgraph(CSET_BUILTIN);
 cset cscntrl(CSET_BUILTIN);
 
-#ifdef isascii
+#if defined(isascii) && (!defined(__DragonFly__) || !defined(__FreeBSD__))
 #define ISASCII(c) isascii(c)
 #else
 #define ISASCII(c) (1)
@@ -86,6 +89,9 @@
   if (initialised)
     return;
   initialised = 1;
+#if defined(__DragonFly__) || defined(__FreeBSD__)
+  (void)setlocale(LC_CTYPE, "");
+#endif
   for (int i = 0; i <= UCHAR_MAX; i++) {
     csalpha.v[i] = ISASCII(i) && isalpha(i);
     csupper.v[i] = ISASCII(i) && isupper(i);
